metricbeat/module/iis/application_pool/reader.go

```diff
@@ -167,7 +170,11 @@ func (re *Reader) fetch(names []string) ([]mb.Event, error) {
 				if val.Instance == appPool.Name {
 					events[appPool.Name].MetricSetFields.Put(appPool.counters[counterPath], val.Measurement)
 				} else if hasWorkerProcess(val.Instance, workers, appPool.WorkerProcessIds) {
-					events[appPool.Name].MetricSetFields.Put(re.WorkerProcesses[counterPath], val.Measurement)
+					if re.WorkerProcesses[counterPath] == ecsProcessId {
```

Comment:
We could also explicitly move the field once the whole event is generated. This is more efficient though :slightly_smiling_face: